src/cpp/core/include/core/http/Keychain.hpp

```diff
@@ -0,0 +1,39 @@
+/*
+ * Keychain.hpp
+ *
+ * Copyright (C) 2020 by RStudio, PBC
+ *
+ * Unless you have received this program directly from RStudio pursuant
+ * to the terms of a commercial license agreement with RStudio, then
+ * this program is licensed to you under the terms of version 3 of the
+ * GNU Affero General Public License. This program is distributed WITHOUT
+ * ANY EXPRESS OR IMPLIED WARRANTY, INCLUDING THOSE OF NON-INFRINGEMENT,
+ * MERCHANTABILITY OR FITNESS FOR A PARTICULAR PURPOSE. Please refer to the
+ * AGPL (http://www.gnu.org/licenses/agpl-3.0.txt) for more details.
+ *
+ */
+
+#ifndef CORE_HTTP_KEYCHAIN_HPP
+#define CORE_HTTP_KEYCHAIN_HPP
+
+#include <vector>
+
+#include <boost/shared_ptr.hpp>
+
+namespace rstudio {
+namespace core {
+namespace http {
+
+struct KeychainCertificateData
+{
+   long size;
+   boost::shared_ptr<unsigned char> data;
+};
+
+std::vector<KeychainCertificateData> getKeychainCertificates();
```

Comment:
Nit: Idiomatically it's more common for us to use a vector of shared pointers to immutable data structures (and using e.g., `std::vector<unsigned char>` inside the data structure to create a managed buffer). This reduces copying somewhat but isn't a big deal here so up to you whether you want to change it.

Reply:
Considering the important thing to free here is the actual underlying (dynamically allocated) buffer, I wanted to wrap that in a `shared_ptr` to ensures it is freed with minimum boilerplate (no custom destructor). So, since the copying overhead here is almost negligible (only an extra 8 bytes per entry), I will leave this as-is.